src/main/java/org/shanerx/tradeshop/Potions.java

```diff
@@ -0,0 +1,299 @@
+        }
+
+        PotionData pd = ((PotionMeta) itm.getItemMeta()).getBasePotionData();
+        String pT = "", t = "", mod = "", pot;
+
+        if (itm.getType() == Material.POTION) {
+            t = "P_";
+        } else if (itm.getType() == Material.SPLASH_POTION) {
+            t = "S_";
+        } else if (itm.getType() == Material.LINGERING_POTION) {
+            t = "L_";
+        }
+
+        if (pd.isExtended()) {
+            mod = "_T";
+        } else if (pd.isUpgraded()) {
+            mod = "_I";
+        }
+
+        switch (pd.getType().toString().toLowerCase()) {
+            case "uncraftable":
+                pT = "EMPTY";
+                break;
+            case "water":
+                pT = "WATER";
+                break;
+            case "thick":
+                pT = "THICK";
+                break;
+            case "awkward":
+                pT = "AWKWARD";
+                break;
+            case "night_vision":
+                pT = "NIGHT";
+                break;
+            case "invisibility":
+                pT = "INVIS";
+                break;
+            case "jump":
+                pT = "LEAP";
+                break;
+            case "speed":
+                pT = "SWIFT";
+                break;
+            case "slowness":
+                pT = "SLOW";
+                break;
+            case "water_breathing":
+                pT = "BREATH";
+                break;
+            case "mundane":
+                pT = "MUNDANE";
+                break;
+            case "fire_resistance":
+                pT = "FIRERES";
+                break;
+            case "instant_heal":
+                pT = "HEALTH";
+                break;
+            case "instant_damage":
+                pT = "HARM";
+                break;
+            case "poison":
+                pT = "POISON";
+                break;
+            case "regen":
+                pT = "REGEN";
+                break;
+            case "strength":
+                pT = "STRENGTH";
+                break;
+            case "weakness":
+                pT = "WEAK";
+                break;
+            case "luck":
+                pT = "LUCK";
+                break;
+        }
+
+        pot = t + pT + mod;
+
+        return pot;
+    }
+
+    public static Potions getPotion(String str) {
+        if (isType(str)) {
+            return valueOf(str.toUpperCase());
+        } else {
+            return null;
+        }
+    }
+
+    public static Potions getPotion(ItemStack itm) {
+        String str = findPotion(itm);
+
+        return getPotion(str);
+    }
+
+    public static boolean isPotion(ItemStack itm) {
+        String str = findPotion(itm);
+        return !(str == null || str.equalsIgnoreCase(""));
+    }
+
+    public static boolean isType(String type) {
+        if (type == null) {
+            return false;
+        }
+
+        try {
+            if (valueOf(type.toUpperCase()) != null) {
```

Comment:
If the null-check already occurs here, what’s the point in repeating it in getPotion()?

Reply:
getPotion() has no null check, it uses this to check for null and illegal arguments, then reutrns null if nothing is found